app/models/request/sample_compound_aliquot_transfer.rb

```diff
@@ -94,31 +51,43 @@ def _aliquots_by_tags_combination
     asset.aliquots.group_by(&:tags_combination)
   end
 
-  def _studies
-    @studies ||= asset.aliquots.map(&:study).uniq
+  def _transfer_into_compound_sample_aliquot(source_aliquots)
+    compound_aliquot = CompoundAliquot.new(request: self, source_aliquots: source_aliquots)
+
+    raise compound_aliquot.errors unless compound_aliquot.valid?
+
+    compound_sample = _create_compound_sample(compound_aliquot)
+
+    _add_aliquot(compound_sample, compound_aliquot)
   end
 
   # Private method to generate a compound sample in a study from a list of
   # component samples
-  def _create_compound_sample(study, component_samples)
+  def _create_compound_sample(compound_aliquot)
+    study = compound_aliquot.default_compound_study
+
     study.samples.create!(
       name: SangerSampleId.generate_sanger_sample_id!(study.abbreviation),
-      component_samples: component_samples
+      component_samples: compound_aliquot.component_samples
     )
   end
 
-  # Default study that the new compound sample will use
-  def _default_compound_study
-    _studies.first
-  end
-
-  # Default project that the new compound sample will use
-  def _default_compound_project_id
-    asset.aliquots.first.project_id
+  def _add_aliquot(sample, compound_aliquot)
+    target_asset
+      .aliquots
+      .create(sample: sample)
```

Comment:
Rather than setting the aliquot attributes individually here, you could do:

```
target_asset.aliquots.create(compound_aliquot.aliquot_attributes)
```

That would let you move a whole chunk of behaviour into compound aliquot and reduce the hits to the database. (This currently hits twice, once on the create, and then on the following save.